en/services/mission.md

```diff
@@ -297,9 +307,59 @@ The protocol has been implemented in C.
 Source code:
 * [src/MissionManager/PlanManager.cc](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/PlanManager.cc)
 
+
 ### ArduPilot
 
-TBD
+ArduPilot implements the mission protocol in C.
+
+ArduPilot uses the same messages and message flow described in this specification. 
+There are some implementation diferences that affect compatibility.
+These are documented below.
+
+Source:
+* [/libraries/GCS_MAVLink/GCS_Common.cpp](https://github.com/ArduPilot/ardupilot/blob/master/libraries/GCS_MAVLink/GCS_Common.cpp)
+
+
+#### Flight Plan Missions
+
+Mission upload, download, clearing missions, and monitoring progress and partial mission upload ([MISSION_WRITE_PARTIAL_LIST](#MISSION_WRITE_PARTIAL_LIST)) are supported.
+
+[Partial mission download](#download_partial) is not supported ([MISSION_REQUEST_PARTIAL_LIST](#MISSION_REQUEST_PARTIAL_LIST)).
+
+ArduPilot's implementation differs from this specification (non-exhaustively):
+- The first mission sequence number (`seq==0`) is populated with the home position of the vehicle instead of the first mission item.
+- Mission uploads are not "atomic". 
+  An upload that fails (or is canceled) part-way through will not match the pre-update state.
+  Instead it may be a mix of the original and new mission.
+- Even if upload is successful, the vehicle mission may not match the version on the uploading system.
+  - If you try and upload more items than ArduPilot can store the system will "accept" the items (i.e. not report a failure) but will just overwrite each new item to the same (highest) slot in the mission list.
+  - Only fields that are used are stored; if an uploaded mission is then downloaded it will not survive the round trip unaltered.
+  
+The following behaviour is not defined by the specification (but is still of interest):
+- ArduPilot performs some validation of fields when mission items are submitted. 
+  The validation code is common to all vehicles; mission items that are not understood by the vehicle type are accepted on upload but skipped during mission execution.
+- A new mission can be uploaded while a mission is being executed.
+  In this case the current waypoint will be executed to completion even if the waypoint sequence is different in the new mission (to get the new item you would need to reset the sequence or switch in/out of auto mode).
+- ArduPilot missions are stored in flash, but run from RAM. 
```

Comment:
This is probably more confusing. It's normally stored in FRAM if your board has FRAM, flash on the small racers that don't, normal storage on a Linux board. It's maybe easier just to say not on an SD card?

Reply:
Sounds good. I have modified to:

> ArduPilot missions are not stored in an SD card and therefore have a vehicle/board-specific maximum mission size (as a benefit, on ArduPilot, missions can survive SD card failure in flight).